Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -758,11 +758,11 @@ - (void)configureForThread:(TSThread *)thread
         [[YapDatabaseViewMappings alloc] initWithGroups:@[ thread.uniqueId ] view:TSMessageDatabaseViewExtensionName];
     [self.uiDatabaseConnection readWithBlock:^(YapDatabaseReadTransaction *transaction) {
       [self.messageMappings updateWithTransaction:transaction];
-      self.page = 0;
-      [self updateRangeOptionsForPage:self.page];
-      [self.collectionView reloadData];
     }];
+    self.page = 0;
```

Comment:
Oh, my intention was simply to move unnecessary work out of the transaction in order to keep transaction lifetime as short as possible.